src/lazy/text/buffer.rs

```diff
@@ -1158,14 +1233,15 @@ impl<'top> TextBufferView<'top> {
         }
         let (remaining, _end_of_eexp) = match whitespace_and_then(tag(")")).parse(remaining) {
             Ok(result) => result,
+            Err(nom::Err::Incomplete(needed)) => return Err(nom::Err::Incomplete(needed)),
             Err(_e) => {
                 return fatal_parse_error(
                     remaining,
                     format!(
-                        "signature has {} parameter(s), e-expression had an extra argument",
-                        signature_params.len()
-                    ),
-                )
+                    "macro {id} signature has {} parameter(s), e-expression had an extra argument",
```

Comment:
Ha! I just made this change on my own branch. Would be handy to show what the extra argument is, too. In my case I had the following macro:
```ion
    (macro CN ($name $sum) (C_1 $name $sum (ONE)))
```
with an invocation like this:
```ion
(:CN "OutstandingRequests"  205e0 )
```

And `arg_expr_cache` as:
```plain
[
  EExpArg { parameter: Parameter { name: "$name", encoding: Tagged, cardinality: ExactlyOne, rest_syntax_policy: NotAllowed }, expr: ValueLiteral(text Ion v1.1 {"OutstandingRequests"}) },
  EExpArg { parameter: Parameter { name: "$sum", encoding: Tagged, cardinality: ExactlyOne, rest_syntax_policy: NotAllowed }, expr: ValueLiteral(text Ion v1.1 {2}) }
]
```

Tracking that down made me think I ought to review this PR :D

Reply:
Yeah, I need to take a pass through the reader and improve all of the error messages. They're too spartan at the moment.